src/sentry/utils/snuba.py

```diff
@@ -669,21 +662,72 @@ def _prepare_query_params(query_params):
     else:
         project_ids = []
 
+    if not project_ids:
+        raise UnqualifiedQueryError(
+            "No project_id filter, or none could be inferred from other filters."
+        )
+
+    organization_id = Project.objects.get(pk=project_ids[0]).organization_id
+
+    return organization_id, {"project": project_ids}
+
+
+def get_query_params_to_update_for_organizations(query_params):
+    """
+    Get the organization ID and query params that need to be updated for organization
+    based datasets, before we send the query to Snuba.
+    """
+    if "org_id" in query_params.filter_keys:
+        organization_ids = list(set(query_params.filter_keys["org_id"]))
+        if len(organization_ids) != 1:
+            raise UnqualifiedQueryError("Multiple organization_ids found. Only one allowed.")
+        organization_id = organization_ids[0]
+    elif "project_id" in query_params.filter_keys:
+        project_ids = list(set(query_params.filter_keys["project_id"]))
+        project = Project.objects.get(pk=project_ids[0])
+        organization_id = project.organization_id
```

Comment:
Why doesn't this just call `get_query_params_to_update_for_projects` which seems to do something very similar?

Reply:
I just wanted to be very explicit about what `query_params.filter_keys` outcomes would accept. So, that it's easy to know that you can only call outcomes on `project_id` and `org_id`. But you do have a point about code duplication. I did:
```
    elif "project_id" in query_params.filter_keys:
        organization_id, _ = get_query_params_to_update_for_projects(query_params)
```

Let me know what you think!

Reply:
Yep, this looks good. 👍